src/sgame/sg_main.cpp

```diff
@@ -651,6 +655,14 @@ void G_InitGame( int levelTime, int randomSeed, bool inClient )
 
 	// Initialize Lua
 	Lua::Initialize();
+
+	if ( !g_luaScript.Get().empty() )
```

Comment:
To make this more useful and less redundant with simply putting a Lua command in a map config, how about delaying the command until the first non-fake frame? There's the server startup code

```
// run a few frames to allow everything to settle
	for ( i = 0; i < GAME_INIT_FRAMES; i++ )
	{
		gvm.GameRunFrame( sv.time );
		svs.time += FRAMETIME;
		sv.time += FRAMETIME;
	}
```

During the first frames some entities are still popping into existence and the floating buildings from Radiant-created layouts have to fall to the ground. It could be nice for the script to wait so that entities will all exist and be in the right place.

Reply:
I have been using the timer to run code on the next frame.

```
Timer.add(1, function() print('next frame') end)
```

Reply:
I didn't mean anything about rerunning the script every frame. I'm saying it may be better not to run the script in the very first frame because the level isn't truly done loading at that point. For example buildables may not exist yet. So it might mess up a script that does something with buildables.